eth/api_tracer.go

```diff
@@ -297,7 +297,9 @@ func (api *PrivateDebugAPI) traceChain(ctx context.Context, start, end *types.Bl
 				database.TrieDB().Reference(root, common.Hash{})
 			}
 			// Dereference all past tries we ourselves are done working with
-			database.TrieDB().Dereference(proot)
+			if proot != (common.Hash{}) {
+				database.TrieDB().Dereference(proot)
```

Comment:
Would it be horribly wrong to add the check against `(common.Hash{})` inside `Dereference`, to prevent removal of the meta-root?

Reply:
We can add it as a warning there.